kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/coordinator/transaction/TransactionMarkerChannelManager.java

```diff
@@ -268,32 +274,80 @@ public void addTxnMarkersToBrokerQueue(String transactionalId,
         for (TopicPartition topicPartition : topicPartitions) {
             String pulsarTopic = new KopTopic(topicPartition.topic(), namespacePrefixForUserTopics)
                     .getPartitionName(topicPartition.partition());
-            CompletableFuture<Optional<InetSocketAddress>> addressFuture =
-                    kopBrokerLookupManager.findBroker(pulsarTopic, sslEndPoint);
             CompletableFuture<Void> addFuture = new CompletableFuture<>();
             addressFutureList.add(addFuture);
-            addressFuture.whenComplete((address, throwable) -> {
-                if (throwable != null) {
-                    log.warn("Failed to find broker for topic partition {}", topicPartition, throwable);
-                    unknownBrokerTopicList.add(topicPartition);
-                    addFuture.completeExceptionally(throwable);
-                    return;
-                }
-                if (!address.isPresent()) {
-                    log.warn("No address for broker for topic partition {}", topicPartition);
-                    unknownBrokerTopicList.add(topicPartition);
-                    addFuture.completeExceptionally(new Exception("no address for owner of " + topicPartition));
-                    return;
-                }
-                addressAndPartitionMap.compute(address.get(), (__, set) -> {
-                    if (set == null) {
-                        set = new ArrayList<>();
-                    }
-                    set.add(topicPartition);
-                    return set;
-                });
-                addFuture.complete(null);
-            });
+            kopBrokerLookupManager.isTopicExists(pulsarTopic)
```

Comment:
isn't this check too heavy ?
we are on the write path here

Reply:
Yes, I also find out. Maybe we can get check if the topic exists for the cache?
```java
org.apache.pulsar.broker.service.BrokerService#fetchPartitionedTopicMetadataAsync
```

Reply:
@Demogorgon314 
Makes sense to me